libyul/optimiser/Suite.cpp

```diff
@@ -104,6 +106,7 @@ void OptimiserSuite::run(
 		VarDeclPropagator{}(ast);
 		RedundantAssignEliminator::run(ast);
 		UnusedPruner::runUntilStabilised(ast, reservedIdentifiers);
+		StructuralSimplifier{}(ast);
```

Comment:
I think it makes sense to run this before the unused pruner, because it could remove some uses of some variables.

Reply:
I moved it to after each run of the expression simplifier now, that's probably a good spot.